api/v1alpha1/runnerdeployment_types.go

```diff
@@ -22,14 +22,17 @@ import (
 
 // RunnerReplicaSetSpec defines the desired state of RunnerDeployment
 type RunnerDeploymentSpec struct {
-	Replicas *int `json:"replicas"`
+	Replicas    *int `json:"replicas"`
+	MinReplicas *int `json:"minReplicas"`
+	MaxReplicas *int `json:"maxReplicas"`
 
 	Template RunnerTemplate `json:"template"`
 }
 
 type RunnerDeploymentStatus struct {
-	AvailableReplicas int `json:"availableReplicas"`
-	ReadyReplicas     int `json:"readyReplicas"`
+	AvailableReplicas int  `json:"availableReplicas"`
+	ReadyReplicas     int  `json:"readyReplicas"`
+	DesiredReplicas   *int `json:"desiredReplicas"`
```

Comment:
Using the `scale` sub-resource here would be a more Kubernetes-like implementation. It looks like controller-runtime already supports the `scale` sub-resource. thoughts?

Reply:
@summerwind Sounds good! But can we address that in another PR? AFAIK, a `scale` sub-resource doesn't make `DesiredReplicas` unnecessary, and the change would be additive.

Reply:
BTW, do you have any opinion on whether we use `Replicas` or `DesiredReplicas` for this field?

Initially, I thought `Replicas` is better, but it turned out that we usually use status.Replicas for something other than what I mean here.

Here's the definition from apps/v1.DeploymentStatus:

```
// DeploymentStatus is the most recently observed status of the Deployment.
type DeploymentStatus struct {
	// The generation observed by the deployment controller.
	// +optional
	ObservedGeneration int64 `json:"observedGeneration,omitempty" protobuf:"varint,1,opt,name=observedGeneration"`

	// Total number of non-terminated pods targeted by this deployment (their labels match the selector).
	// +optional
	Replicas int32 `json:"replicas,omitempty" protobuf:"varint,2,opt,name=replicas"`

	// Total number of non-terminated pods targeted by this deployment that have the desired template spec.
	// +optional
	UpdatedReplicas int32 `json:"updatedReplicas,omitempty" protobuf:"varint,3,opt,name=updatedReplicas"`

	// Total number of ready pods targeted by this deployment.
	// +optional
	ReadyReplicas int32 `json:"readyReplicas,omitempty" protobuf:"varint,7,opt,name=readyReplicas"`

	// Total number of available pods (ready for at least minReadySeconds) targeted by this deployment.
	// +optional
	AvailableReplicas int32 `json:"availableReplicas,omitempty" protobuf:"varint,4,opt,name=availableReplicas"`

	// Total number of unavailable pods targeted by this deployment. This is the total number of
	// pods that are still required for the deployment to have 100% available capacity. They may
	// either be pods that are running but not yet available or pods that still have not been created.
	// +optional
	UnavailableReplicas int32 `json:"unavailableReplicas,omitempty" protobuf:"varint,5,opt,name=unavailableReplicas"`

	// Represents the latest available observations of a deployment's current state.
	// +patchMergeKey=type
	// +patchStrategy=merge
	Conditions []DeploymentCondition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,6,rep,name=conditions"`

	// Count of hash collisions for the Deployment. The Deployment controller uses this
	// field as a collision avoidance mechanism when it needs to create the name for the
	// newest ReplicaSet.
	// +optional
	CollisionCount *int32 `json:"collisionCount,omitempty" protobuf:"varint,8,opt,name=collisionCount"`
}
```

It does say that `status.replicas` is for `Total number of non-terminated pods targeted by this deployment (their labels match the selector).`, which implies to include outdated pods as well.

In contrast, RunnerDeploymnet's status.desiredReplicas is basically `total number of desired and non-terminated pods to be set for the primary runnerset`.

Reply:
Thanks. Changed to `Replicas` in 0d45cef 😃